config/initializers/sphinx.rb

```diff
@@ -20,3 +20,19 @@ def total_pages
     @total_pages ||= ([total_entries, 1000].min / search.per_page.to_f).ceil
   end
 end)
+
+# implement conditionally inserting or deleting from the index
+# see https://github.com/pat/thinking-sphinx/pull/1258
+ThinkingSphinx::Processor.include(Module.new do
+  def stage
+    real_time_indices.each do |index|
+      found = index.scope.find_by(model.primary_key => id)
+
+      if found
+        ThinkingSphinx::RealTime::Transcriber.new(index).copy found
+      else
+        ThinkingSphinx::Deletion.perform(index, id)
+      end
+    end
+  end
+end)
```

Comment:
I read the linked PR and now I'm even more confused. Could you please provide a global picture on what is this for?

Reply:
When object is found in AR scope, it is indexed in manticore, otherwise it is removed from index.